mcfix_delivery/models/stock_picking.py

```diff
@@ -0,0 +1,15 @@
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.multi
+    @api.constrains('company_id', 'carrier_id')
```

Comment:
Maybe it should be better to also add an onchange and put carrier_id = False, so no errors are raised